overlord/state/task.go

```diff
@@ -53,24 +54,26 @@ func newTask(state *State, id, kind, summary string) *Task {
 }
 
 type marshalledTask struct {
-	ID       string                      `json:"id"`
-	Kind     string                      `json:"kind"`
-	Summary  string                      `json:"summary"`
-	Status   Status                      `json:"status"`
-	Progress progress                    `json:"progress"`
-	Data     map[string]*json.RawMessage `json:"data"`
+	ID         string                      `json:"id"`
+	Kind       string                      `json:"kind"`
+	Summary    string                      `json:"summary"`
+	Status     Status                      `json:"status"`
+	Progress   progress                    `json:"progress"`
+	Data       map[string]*json.RawMessage `json:"data"`
+	WaitingFor []*Task                     `json:"waiting-for"`
```

Comment:
Same (`waitTasks` and `wait-tasks`), and we also want this to be a list of task ids, otherwise we'll have multiple copies of the same task in memory once this roundtrips through the disk, and we'll never see it updated.

It's probably easier to store just the task ids in Task.waitTasks as well, and map these back into the actual values inside WaitTasks, which is the only way to reach it from the outside.